### PR TITLE
 Update the version of library dependencies

### PR DESCRIPTION
**Pull request by perfah**
Such as:

- git submodule?
- Update bindgen to latest version
- Refactor deprecated function calls (such as 'bindgen::builder::whitelisted_type(...)')
- Remove references to protocols that no longer exists (such as 'wlroots/protocol/screenshooter.xml')

Make sure everything compiles.

### CODE REVIEW

**Comment by perfah** [comment]
This looks completed! Is it? Would be good merge it for the rest of the issues.

**Comment by perfah** [comment]
Maybe we also should update this:
`wayland-scanner = "0.21.*"`
to
`wayland-scanner = "0.25.*"`

**Comment by gynther-k** [comment]
@perfah When I try to upgrade to `wayland-scanner = "0.25.*"  ` I get som errors that seems to be fixable, but this one I don't know:
```
error[E0425]: cannot find function `generate_c_interfaces` in crate `wayland_scanner`
   --> wlroots-sys/build.rs:209:26
    |
209 |         wayland_scanner::generate_c_interfaces(
    |                          ^^^^^^^^^^^^^^^^^^^^^ not found in `wayland_scanner`
```

Even though they write about generate_c_interfaces
in https://docs.rs/crate/wayland-scanner/0.25.0

**Comment by perfah** [comment]
It could be worthwhile to update wayland-scanner and fix these errors. I however can't get issue/1 to build with the unstable feature at the moment.

**Comment by gynther-k** [comment]
@perfah I had to use this procedure for the different builds. But I got some errors that you probably fixed with the latest pushes.

```
git clone https://github.com/perfah/wlroots-rs.git
git checkout issue/1
git submodule update --init
cd wlroots-sys

sudo apt-get install 
cargo 
pkg-config 
wayland-protocols 
libwayland-bin 
clang 
llvm 
libpixman-1-dev 
libwayland-dev 

cargo build 

```
```
cargo build --features unstable

libudev-dev
libxkbcommon-dev
libgles2-mesa-dev
libinput-dev 
libinput-bin
```

**Comment by perfah** [comment]
Yep, the commits fixed the building with the unstable-feature. I just tried updating wayland-scanner but it looks like the documentation is misleading as you pointed out. I have no idea why they would remove  wayland_scanner::generate_c_interfaces but I think it seems pointless to update it at the moment.

**Comment by gynther-k** [comment]
@perfah it is discussed here https://github.com/Smithay/wayland-rs/issues/302

**Comment by perfah** [comment]
I tried using the wayland_scanner::generate_code call yesterday, but it resulted in 146 errors in other parts of the code.

Edit: I fixed these errors in 74fd086 and updated all wayland-* libraries.

File: wlroots-sys/build.rs [hunk 1]
```diff
@@ -192,24 +192,24 @@ fn generate_protocols() {
...
             output_dir.join(format!("{}_client_api.rs", protocol.1)),
             wayland_scanner::Side::Client
         );
-        wayland_scanner::generate_c_interfaces(
+        wayland_scanner::generate_code(
             protocol.0,
-            output_dir.join(format!("{}_interfaces.rs", protocol.1))
+            output_dir.join(format!("{}_interfaces.rs", protocol.1)),
+            wayland_scanner::Side::Server
         );
```
Comment (elinorbgr) [inline wlroots-sys/build.rs:213]: Sorry for reacting only now, but with the new version of `wayland-scanner` you don't actually need the `*_interfaces.rs` files any longer. So you can remove this part of the build script as well as the `c_interfaces` module in your library, they are no longer used.

I've now updated the documentation for wayland-scanner, it is not yet published, but is available here: https://smithay.github.io/wayland-rs/wayland_scanner/

Comment (perfah) [reply wlroots-sys/build.rs:213]: @vberger Thanks for clearing it up! Very appreciated :+1:

**Pull request merged**